workspace.go

```diff
@@ -203,6 +204,11 @@ type VCSRepo struct {
 	WebhookURL        string `jsonapi:"attr,webhook-url"`
 }
 
+type WorkspaceSettingOverwrites struct {
+	ExecutionMode *bool `jsonapi:"attr,execution-mode,omitempty"`
+	AgentPool     *bool `jsonapi:"attr,agent-pool,omitempty"`
```

Comment:
Right, so:

- `WorkspaceSettingOverwrites` represents the values in a Workspace struct that we received _from_ the API. 
- `WorkspaceSettingOverwritesOptions` represents the values we may or may not be sending _to_ the API in a create or update options payload. 

Thus, this struct (the value one) should use plain `bool`s, and should not be using `omitempty`.

...I think! The theory here is that when you get a value back from the API, it should ALWAYS have either a true or false value for each of these, so we reflect that in the type system by not allowing a nil pointer as a third option. But, push back and correct me if I'm wrong in that — for example, I haven't yet tried running against an old TFE that doesn't provide this field (or doesn't provide a specific field in it, which is something we'll run into as the struct grows).

For prior art, see the nested VCSRepo / VCSRepoOptions structs, which are also doing this exact thing.

Reply:
I think that's true, EXCEPT if we end up adding new values to this object in future updates. Presumably, folks then modify the old values to be pointers as well for the sake of consistency. I'm assuming that's why in `VCSRepoOptions` today, the values are all pointers with `omitempty`.

I'm happy to modify the `ExecutionMode` and `AgentPool` values to be included today, since I think you are TECHNICALLY correct, but, unless I'm mistaken, it seems like that will lead to inconsistencies when someone adds the next "setting-overwrites". OR worse yet, they'll think they don't need the `omitempty` in which case, their code will break on older versions of TFE.

Reply:
For the lurkers out there: Nick and I chatted about this in the #team-tf-cli slack channel. The end result is we'd like to use pointers SOLELY in the interest of making supporting backwards-compatibility easier. By allowing these fields to be pointers, users can check to see if the attributes are `nil` or `false`, which can give go-tfe insight into what version of TFE it is interacting with.